cmd/epp/runner/runner.go

```diff
@@ -467,6 +470,33 @@ func (r *Runner) parseConfigurationPhaseTwo(ctx context.Context, rawConfig *conf
 	return cfg, nil
 }
 
+func (r *Runner) deprecatedFlagsHandler(logger logr.Logger) error {
+	deprecated := map[string]string{ // Map of deprecated flags: key = flag name, value = replacement (empty if none)
+		"model-server-metrics-port": "",
+	}
+
+	var usedDeprecated []string // Track which deprecated flags were explicitly set
+	flag.Visit(func(f *flag.Flag) {
+		if _, ok := deprecated[f.Name]; ok {
+			usedDeprecated = append(usedDeprecated, f.Name)
+		}
+	})
+
+	if len(usedDeprecated) > 0 { // report on deprecated flags used
+		for _, option := range usedDeprecated {
+			if replacement := deprecated[option]; replacement != "" {
+				logger.Info("deprecated option will be removed in the next release",
+					"option", option, "replacement", replacement)
+			} else {
+				logger.Info("deprecated option will be removed in the next release. No replacement available.",
+					"option", option)
+			}
+		}
+		return errors.New("deprecated options used" + fmt.Sprintf("%+v", usedDeprecated))
+	}
+	return nil
+}
```

Comment:
this function can be simplified to a 5 liner.
few points:
- as for now, we have no depracated flag that is replaced by a different flag. so the replacement part seems redundant.
- no need to use map for deprecated (can be a set).
- no need to run flag.Visit and then iterating over it again. while we're at the flag.Visit, we can check - if the depracated flag is used, log the deprecation notice.
- no need to return error (which is later ignored anyway).

Reply:
Correct. Was laying down work for future use but can hold on it until it is needed.
Replaced with explicit handling of the one flag.